pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisRecordProcessor.java

```diff
@@ -112,4 +79,45 @@ private void checkpoint(IRecordProcessorCheckpointer checkpointer) {
         }
     }
 
+    @Override
+    public void initialize(InitializationInput initializationInput) {
+        kinesisShardId = initializationInput.shardId();
+    }
+
+    @Override
+    public void processRecords(ProcessRecordsInput processRecordsInput) {
+
+        log.info("Processing " + processRecordsInput.records().size() + " records from " + kinesisShardId);
+
+        for (KinesisClientRecord record : processRecordsInput.records()) {
+            try {
+                queue.put(new KinesisRecord(record));
+            } catch (InterruptedException e) {
+                log.warn("unable to create KinesisRecord ", e);
+            }
+        }
+
+        // Checkpoint once every checkpoint interval.
+        if (System.nanoTime() > nextCheckpointTimeInNanos) {
+            checkpoint(processRecordsInput.checkpointer());
+            nextCheckpointTimeInNanos = System.nanoTime() + checkpointInterval;
+        }
+    }
+
+    @Override
+    public void leaseLost(LeaseLostInput leaseLostInput) {
+        log.info("lease lost, will terminate soon");
```

Comment:
Seems strange to not do anything with the leaseLostInput object here

Reply:
the KCL docs seem to indicate we will exit if we get here, hence the log statement